metal/classifier.py

```diff
@@ -461,80 +489,127 @@ def _set_optimizer(self, train_config):
                     "SparseAdam optimizer does not support weight_decay (l2 penalty)."
                 )
         else:
-            raise ValueError(f"Did not recognize optimizer option '{opt}''")
-        return optimizer
+            raise ValueError(f"Did not recognize optimizer option '{opt}'")
+        self.optimizer = optimizer
 
-    def _set_scheduler(self, scheduler_config, optimizer):
-        scheduler = scheduler_config["scheduler"]
-        if scheduler is None:
+    def _set_scheduler(self, train_config):
+        lr_scheduler = train_config["lr_scheduler"]
+        if lr_scheduler is None:
             lr_scheduler = None
-        elif scheduler == "exponential":
-            lr_scheduler = torch.optim.lr_scheduler.ExponentialLR(
-                optimizer, **scheduler_config["exponential_config"]
-            )
-        elif scheduler == "reduce_on_plateau":
-            lr_scheduler = torch.optim.lr_scheduler.ReduceLROnPlateau(
-                optimizer, **scheduler_config["plateau_config"]
-            )
         else:
-            raise ValueError(
-                f"Did not recognize scheduler option '{scheduler}''"
-            )
-        return lr_scheduler
+            lr_scheduler_config = train_config["lr_scheduler_config"]
+            if lr_scheduler == "exponential":
+                lr_scheduler = torch.optim.lr_scheduler.ExponentialLR(
+                    self.optimizer, **lr_scheduler_config["exponential_config"]
+                )
+            elif lr_scheduler == "reduce_on_plateau":
+                lr_scheduler = torch.optim.lr_scheduler.ReduceLROnPlateau(
+                    self.optimizer, **lr_scheduler_config["plateau_config"]
+                )
+            else:
+                raise ValueError(
+                    f"Did not recognize lr_scheduler option '{lr_scheduler}'"
+                )
+        self.lr_scheduler = lr_scheduler
 
-    def score(
+    # TEMPORARY FOR SLICING PROJECT; there may be a better way to do this
```

Comment:
Should this be included on master?  Probably not?

Reply:
Nope! Good catch. Removed.